stan/math/opencl/prim/neg_binomial_2_log_glm_lpmf.hpp

```diff
@@ -82,19 +83,22 @@ return_type_t<T_alpha, T_beta, T_precision> neg_binomial_2_log_glm_lpmf(
     check_size_match(function, "Rows of ", "x_cl", N, "size of ", "alpha",
                      stan::math::size(alpha));
   }
-  check_positive_finite(function, "Precision parameter", phi);
+  const auto& phi_ref = to_ref<Eigen::Stride<0, 0>>(phi);
```

Comment:
Default Stride assumes contiguous memory on inner dimension (that is columns in column-major matrix), but not necessary contiguous on outer dimension. If `Eigen::Ref` or `to_ref` is constructed from expression that violates this assumption it will be evaluated (for example transpose of a matrix would be evaluated but block of a matrix would not)

To copy data to OpenCL device we need contiguous memory in both dimensions or we need to make a copy of data within the constructor of `matrix_cl`. Instead we can ensure that when calling `to_ref`.

Huh I just remembered we only have col-major matrices on OpenCL devices, so I will also have to ensure that part.

Reply:
I will just revert this for now, and put these OpenCL-related details in a separate PR.

Reply:
Got it, sounds good